repo/manifest/committed_manifest_manager.go

```diff
@@ -258,15 +247,19 @@ func (m *committedManifestManager) ensureInitialized(ctx context.Context) error
 	m.lock()
 	defer m.unlock()
 
-	if m.initialized {
+	rev := m.b.Revision()
```

Comment:
It may be worth renaming `ensureInitialized` to `maybeRefresh` or something along those lines